api/resolvers/dmarc_report.py

```diff
@@ -0,0 +1,102 @@
+from graphql import GraphQLError
+
+from db import db
+
+from functions.input_validators import cleanse_input
+from functions.auth_wrappers import require_token
+from functions.auth_functions import is_super_admin, is_user_read
+
+from models import Domains, Dmarc_Reports
+
+from schemas.domain.dmarc_report import DmarcReport
+
+
+@require_token
+def resolve_dmarc_reports(self: DmarcReport, info, **kwargs):
+    """
+
+    :param self:
+    :param info:
+    :param kwargs:
+    :return:
+    """
+    user_id = kwargs.get('user_id')
+    user_roles = kwargs.get('user_roles')
+    domain = cleanse_input(kwargs.get('domain'))
+    start_date = kwargs.get('start_date')
+    end_date = kwargs.get('end_date')
+
+    if (start_date is not None and end_date is None) or \
+        (start_date is None and end_date is not None):
+        raise GraphQLError('Error, both start and end dates are required.')
```

Comment:
The annotated file stated that `end_date` is far more important than `start_date`.  Do we think it is necessary to strictly enforce that both are required?

Reply:
This is for selecting a date range, so I need two times to select between ?

Reply:
Ahh yes,  I see that now!  Looks good to me!! :smile: